src/components/FavoriteButton.tsx

```diff
@@ -0,0 +1,69 @@
+import { useFocusEffect } from '@react-navigation/native'
+import React, { ReactElement } from 'react'
+import { widthPercentageToDP as wp } from 'react-native-responsive-screen'
+import styled from 'styled-components/native'
+
+import { StarCircleIconGrey, StarCircleIconGreyFilled } from '../../assets/images'
+import { Document } from '../constants/endpoints'
+import labels from '../constants/labels.json'
+import useLoadAsync from '../hooks/useLoadAsync'
+import AsyncStorage from '../services/AsyncStorage'
+import { reportError } from '../services/sentry'
+
+const Container = styled.View`
+  padding: ${props => `${props.theme.spacings.xs} 0  ${props.theme.spacings.xs} ${props.theme.spacings.sm}`};
+  align-self: center;
+`
+
+const Icon = styled(StarCircleIconGreyFilled)`
+  min-width: ${wp('9%')}px;
+  min-height: ${wp('9%')}px;
+`
+const IconOutline = styled(StarCircleIconGrey)`
+  min-width: ${wp('9%')}px;
+  min-height: ${wp('9%')}px;
+`
+const Button = styled.TouchableOpacity`
+  justify-content: center;
+  align-items: center;
+  shadow-radius: 5px;
+  shadow-offset: 1px 1px;
+  shadow-opacity: 0.5;
+`
+
+interface Props {
+  document: Document
+  refreshFavorites?: () => void
+}
+
+const FavoriteButton = ({ document, refreshFavorites }: Props): ReactElement | null => {
+  const { data: isFavorite, refresh } = useLoadAsync(AsyncStorage.isFavorite, document)
```

Comment:
Why refresh and refreshFavorites?

Reply:
One refreshes the button and one the favorites list which is sadly both needed I think. Do you have another solution?

Reply:
maybe renaming could make it clearer. E.g. refresh as it is, an refreshParent oder refreshList or something?

Reply:
Renamed to `onFavoritesChanged` as it doesn't matter if it refreshes or does something entirely else.